scripts/update-filter.js

```diff
@@ -0,0 +1,107 @@
+    const [filterStr, dappStr] = [`${filterName}@${filter}`, `${dappName}@${dapp}`];
+    if (filter === ethers.constants.AddressZero) {
+      const newFilter = await filterDeployer();
+      console.log(`Adding ${filterName}@${newFilter} for ${dappStr}`);
+      await dappRegistry.addDapp(registryId, dapp, newFilter);
+      console.log(`Done. Filter will be active on ${new Date(Date.now() + timelock).toLocaleString()}\n`);
+    } else {
+      const pendingUpdate = await dappRegistry.pendingFilterUpdates(registryId, dapp);
+      const pendingUpdateConfirmationTime = 1000 * parseInt(new BN(pendingUpdate.slice(2), 16).maskn(64).toString(16), 16);
+      const pendingUpdateFilterAddress = `0x${pendingUpdate.slice(10, 50)}`;
+      if (pendingUpdate === ethers.constants.HashZero) {
+        console.log(`Existing filter ${filterStr} found for ${dappStr}. Press any key to confirm its replacement, or Cmd+C to cancel.\n`);
+        await keypress();
+        const newFilter = await filterDeployer();
+        console.log(`Requesting replacement of ${filterStr} by ${filterName}@${newFilter} for ${dappStr}`);
+        await dappRegistry.requestFilterUpdate(registryId, dapp, newFilter);
+        console.log(`Done. Pending filter update will be confirmable on ${new Date(Date.now() + timelock).toLocaleString()}\n`);
+      } else if (Date.now() < pendingUpdateConfirmationTime) {
+        const confTime = new Date(pendingUpdateConfirmationTime).toLocaleString();
+        console.log(`Pending installation of ${filterName}@${pendingUpdateFilterAddress} for ${dappStr} will be confirmable on ${confTime}\n`);
+      } else {
+        console.log(`Confirming installation of ${filterName}@${pendingUpdateFilterAddress} for ${dappStr}`);
+        await dappRegistry.confirmFilterUpdate(registryId, dapp);
+        console.log("Done.\n");
+      }
+    }
+  };
+
+  /////////////////////////////////////////////////////////////////////////////
+  //
+  // Add a call to installFilter for every filter that you want to update:
+  //
+  // await installFilter({
+  //   filterDeployer: async () => {
+  //     const myFilterToUpdate = await MyFilterToUpdate.deploy(...);
+  //     console.log(`Deployed MyFilterToUpdate at ${myFilterToUpdate.address}`);
+  //     configUpdate.myDapp.filter = myFilterToUpdate.address;
+  //     return uniV3RouterFilter.address;
+  //   },
+  //   dapp: config.myDapp.address,
+  //   dappName: "MyDapp",
+  //   filterName: "MyDappFilter",
+  // });
+  //
+  /////////////////////////////////////////////////////////////////////////////
+
+  // Give ownership back
+  if (registryOwner != deployer.address) {
+    await dappRegistry.changeOwner(TRUSTLIST, registryOwner);
+  }
+
+  // update config
+  await configLoader.save(configUpdate);
+  }
+
+  main()
+      .then(() => process.exit(0))
+      .catch((error) => {
+      console.error(error);
+      process.exit(1);
+      });
```

Comment:
It's possible to use async/await syntax in top level code by wrapping in an IIFE:

```suggestion
(async () => {
  try {
    await main();
    process.exit(0);
  } catch (error) {
    console.error(error);
    process.exit(1);
  }
})();
```